app/services/CategorisationService.scala

```diff
@@ -16,216 +16,124 @@
-                                  getGoodsRecordResponse.countryOfOrigin,
-                                  LocalDate.now()
-                                )
-      originalCommodityCode   = originalCommodityCodeOpt.getOrElse(getGoodsRecordResponse.comcode)
-      categorisationInfo     <- CategorisationInfo.build(goodsNomenclature, Some(originalCommodityCode)) match {
-                                  case Some(categorisationInfo) => Future.successful(categorisationInfo)
-                                  case _                        => Future.failed(new RuntimeException("Could not build categorisation info"))
-                                }
-      updatedAnswers         <-
-        Future.fromTry(
-          request.userAnswers.set(
-            RecordCategorisationsQuery,
-            recordCategorisations.copy(records = recordCategorisations.records + (recordId -> categorisationInfo))
-          )
-        )
-      _                      <- sessionRepository.set(updatedAnswers)
-    } yield updatedAnswers
   }
 
-  def updateCategorisationWithUpdatedCommodityCode(
-    request: DataRequest[_],
+  def calculateResult(
+    categorisationInfo: CategorisationInfo,
+    userAnswers: UserAnswers,
     recordId: String
-  )(implicit
-    hc: HeaderCarrier
-  ): Future[UserAnswers] = {
-
-    val recordCategorisations =
-      request.userAnswers.get(RecordCategorisationsQuery).getOrElse(RecordCategorisations(Map.empty))
-
-    for {
-      newCommodityCode                 <- Future.fromTry(Try(request.userAnswers.get(CommodityUpdateQuery(recordId)).get))
-      getGoodsRecordResponse           <- goodsRecordsConnector.getRecord(eori = request.eori, recordId = recordId)
-      goodsNomenclature                <- ottConnector.getCategorisationInfo(
-                                            newCommodityCode.commodityCode,
-                                            request.eori,
-                                            request.affinityGroup,
-                                            Some(recordId),
-                                            getGoodsRecordResponse.countryOfOrigin,
-                                            LocalDate.now()
-                                          )
-      categorisationInfo               <- Future.fromTry(Try(CategorisationInfo.build(goodsNomenclature).get))
-      updatedAnswers                   <-
-        Future.fromTry(
-          request.userAnswers.set(
-            RecordCategorisationsQuery,
-            recordCategorisations.copy(records = recordCategorisations.records + (recordId -> categorisationInfo))
-          )
-        )
-      updatedAnswersCleanUpAssessments <-
-        Future.fromTry(cleanupOldAssessmentAnswers(updatedAnswers, recordId))
-      _                                <- sessionRepository.set(updatedAnswersCleanUpAssessments)
-    } yield updatedAnswersCleanUpAssessments
-  }
+  ): Scenario =
+    if (categorisationInfo.categoryAssessments.isEmpty) {
```

Comment:
Could kind of 'flatten' it with pattern matching if you think it's worth it: (DON'T ASSUME THIS CODE IS CORRECT)
```
categorisationInfo match {
  case info if info.categoryAssessments.isEmpty =>
    StandardGoodsNoAssessmentsScenario

  case info 
    if info.categoryAssessmentsThatNeedAnswers.isEmpty && 
       info.categoryAssessments.exists(_.isCategory1) =>
    Category1NoExemptionsScenario

  case info if info.categoryAssessmentsThatNeedAnswers.isEmpty =>
    Category2Scenario

  case _ =>
    calculateBasedOnAnswers(categorisationInfo, userAnswers, recordId)
}
```

Reply:
I don't think that's any nicer now I look at it actually :')

Reply:
That could work. Would hold off for now just because its one of two places that need to change for Niphls/Nirms which will mean it changes soon anyway :)